### DISCUSSION TITLE
Best practices for managing data property names case  (i.e. snake_case to camelCase)

### DISCUSSION THREAD

**Discussion by andreyvolokitin** [algolia/instantsearch#5049] [Q&A] [answered]
In `autocomplete` we have `reshape` which can be used for these purposes. For `InstantSearch` it might be helpful to convert `hits` similar to items in autocomplete, what are the best practices for doing this in `InstantSearch`?

**Comment by Haroenv** [comment, marked as answer]
hits has `transformItems` that can be used to change the shape. The docs are available here: https://www.algolia.com/doc/api-reference/widgets/hits/js/#widget-param-transformitems

If you want more advanced grouping of the results or similar, you may want to use the connector [`connectHits`](https://www.algolia.com/doc/api-reference/widgets/hits/js/#connector)

Is that what you were looking for @andreyvolokitin ?

**Reply by andreyvolokitin** [reply]
Yeah, this will do it, is this the "highest spot" to transform data? I.e. if we could transform it "at the root" somewhere this would allow to not repeat the transformation for each widget. The purpose is to convert snake_case to camelCase for data property names (in order to use more conventional snakeCase at the frontend)

**Reply by andreyvolokitin** [reply]
I think `search` method for the client might be "the root"? I.e. if we convert hits there, would it cause any problems, like if some `instantsearch` internals would depend on properties being in the original case? For the `autocomplete` I transformed all the properties but ignored any properties starting with `_` to leave internal Algolia properties intact

**Reply by Haroenv** [reply]
yes, if you want to globally change the attributes, it's best to change both in the parameters and response of the search method of the search client, like in this guide: https://www.algolia.com/doc/guides/building-search-ui/going-further/conditional-requests/js/
